### PR TITLE
Interfaces for docker volumes

### PR DESCRIPTION
**Pull request by Jimbo4794**
Signed-off-by: James Davies <jdavies47@hotmail.co.uk>

### CODE REVIEW

**Comment by Jimbo4794** [comment]
Will fix the defaulting error upon implementation

File: galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/DockerContainer.java [hunk 1]
```diff
@@ -70,4 +70,13 @@
...
     public String dockerEngineTag() default "PRIMARY";
 
+    /**
+     * The <code>volumes</code> will be used to indicate that a volume is to be mounted to the container when the continer is creared. The value of this
+     * field can be a comma seperated list if multiple volumes are to be mounted. The general format of the value should look like 
+     * <code>volumes = "volume_name_a:volume_mount_path, volume_name_b:volume_mount_path"</code>. This will search the docker engine for a volume that 
+     * has this name, and create one if it doesn't.
+     * @return
+     */
+    public String volumes();
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/DockerContainer.java:80]: Should be an array of Strings at the very least.
Actually,   the volumes should be an array of DockerVolume annotations. with the volumename and the path as invidual strings.
You will need to indicate that the volume will be prefixed with the container/runname to ensure the volume names do not clash with other test runs

File: galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/IDockerVolume.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+public interface IDockerVolume {
+
+    /**
+     * To be called from a containerCreate method. Provides the json object required for 
+     * mounting a volume to the container.
+     * 
+     * @return JsonObject
+     * @throws DockerManagerException 
+     */
+    public JsonObject getMountJson() throws DockerManagerException;
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/IDockerVolume.java:28]: Dont return the JSON.   Return a parsed structure to insulate the tester from changes to the docker engine.
Only provide the variables that you think the tester needs to know.

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]
I don't see what these volumes gives the test from these interfaces.   What is the point of creating a volume?????
Docker containers do need a volume to operate, even if defined in the Dockerfile.  The reason for the volume would be to share across containers and to preserve data after a container is removed.
Think you need to ask the enhancement requestor for the reasons behind the request.

File: galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/DockerContainer.java (hunk 1 above)
Comment (Michael-Baylis) [reply galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/DockerContainer.java:80]: prefixed with GALASA_runname_,  leave the containername off of it incase the volume is shared across containers.
GALASA being the same prefix as the containers

**Comment by Jimbo4794** [comment]
No longer relevant

**Pull request closed without merging**